map.go

```diff
@@ -935,13 +935,49 @@ func (m *Map) BatchLookupAndDelete(prevKey, nextKeyOut, keysOut, valuesOut inter
 	return m.batchLookup(sys.BPF_MAP_LOOKUP_AND_DELETE_BATCH, prevKey, nextKeyOut, keysOut, valuesOut, opts)
 }
 
+func (m *Map) batchLookupPerCPU(cmd sys.Cmd, count int, startKey, nextKeyOut, keysOut, valuesOut interface{}, opts *BatchOptions) (int, error) {
+	valueBuf := make([]byte, count*int(m.fullValueSize))
```

Comment:
This might maybe not directly related to the proposed change, but to reduce the number if allocations it would be great if the user could provide this slice and potentially reuse it.

Reply:
It's nice that at least this buffer is only ever held on the stack so should be easy to collect?

Reply:
```
map.go-1007-func (m *Map) batchLookup(cmd sys.Cmd, cursor *BatchCursor, keysOut, valuesOut interface{}, opts *BatchOptions) (int, error) {
...
map.go:1044:    valueBuf := make([]byte, count*int(m.fullValueSize))
map.go:1045:    valuePtr := sys.NewSlicePointer(valueBuf)
```
```
./map.go:1044:18: make([]byte, count * int(m.fullValueSize)) escapes to heap
```
```
map.go-1059-func (m *Map) batchLookupPerCPU(cmd sys.Cmd, count int, inBatch []byte, cursor *BatchCursor, keysOut, valuesOut interface{}, opts *BatchOptions) (int, error) {
map.go:1060:    valueBuf := make([]byte, count*int(m.fullValueSize))
map.go:1061:    valuePtr := sys.NewSlicePointer(valueBuf)
```
```
./map.go:1060:18: make([]byte, count * int(m.fullValueSize)) escapes to heap
```